wallet/wallet.go

```diff
@@ -1989,11 +1989,19 @@ type GetTicketsResult struct {
 	Tickets []*TicketSummary
 }
 
-// GetTickets returns a list of TicketSummarys that are located in between the
+// GetTickets calls function f for all tickets located in between the
 // given startBlock and endBlock.  TicketSummary includes TransactionSummmary
 // for the ticket and the spender (if already spent) and the ticket's current
-// status.
-func (w *Wallet) GetTickets(ctx context.Context, chainClient *dcrrpcclient.Client, startBlock, endBlock *BlockIdentifier) (*GetTicketsResult, error) {
+// status. The function f also receives block metadata of the ticket. All
+// tickets on a given call belong to the same block and at least one ticket
+// is present when f is called.
+//
+// The function f may return an error which, if non-nil, is propagated to the
+// caller.  Additionally, a boolean return value allows exiting the function
+// early without reading any additional transactions when true.
+//
+// The arguments to f may be reused and should not be kept by the caller.
+func (w *Wallet) GetTickets(f func([]*TicketSummary, *udb.BlockMeta) (bool, error), chainClient *dcrrpcclient.Client, startBlock, endBlock *BlockIdentifier) error {
```

Comment:
Can we change this to pass the block header instead of a udb.BlockMeta?  The udb types are internal and I don't want them exposed from the wallet's public api if possible.

Reply:
We should also allow nil to be passed in the case that it's an unmined ticket.  The current version will pass a zero-value udb.BlockMeta which is unexpected.